File: ios/MochiApp/DesignSystem/Theme.swift
import SwiftUI

enum MochiColor {
    static let purple = Color(red: 0.545, green: 0.361, blue: 0.965)
    static let purpleDark = Color(red: 0.486, green: 0.227, blue: 0.929)
    static let pink = Color(red: 0.925, green: 0.286, blue: 0.600)
    static let pinkLight = Color(red: 0.976, green: 0.716, blue: 0.855)
    static let lavender = Color(red: 0.808, green: 0.749, blue: 0.976)

    /// Pure black. This was a soft dark grey for a while, chosen by eye off a screenshot; sampling
    /// the glyph cores in docs/figma/1.png settles it — section headers, card titles, card
    /// subtitles, carousel names, pill labels, button labels and "see all" all come back exactly
    /// (0, 0, 0). Text that reads lighter in the design is lighter because of its Inter weight,
    /// not its colour.
    static let textPrimary = Color.black
    static let textSecondary = Color(red: 0.46, green: 0.44, blue: 0.50)

    /// Flat #AAAAAA. Sampled off docs/figma/2.png (Community): the search placeholder,
    /// "24 Themes" under a creator name and every theme description come back as the same
    /// neutral grey — not a desaturated purple, and not `textSecondary`.
    static let textMuted = Color(red: 170 / 255, green: 170 / 255, blue: 170 / 255)

    /// #9750AB — the "by <creator>" byline under a Top Themes card. Lighter and pinker than
    /// `logoSolid` (#9C28B1), which is what the verified-seal badge uses.
    static let creatorLink = Color(red: 151 / 255, green: 80 / 255, blue: 171 / 255)

    /// #F44336 — the filled like heart. Material Red 500, not the app's own `pink`.
    static let heart = Color(red: 244 / 255, green: 67 / 255, blue: 54 / 255)

    /// #8A4FA0 — 1pt outline on the Community search field, the filter pills and the round
    /// download buttons. The white cards themselves carry no stroke at all, only a soft shadow.
    static let outline = Color(red: 138 / 255, green: 79 / 255, blue: 160 / 255)

    /// Pixel-sampled directly off docs/figma/1.png (RGB 156,40,177 = #9C28B1) — the "Mochi"
    /// wordmark is a flat solid color, not a gradient; matches android's MochiColor.logoSolid.
    static let logoSolid = Color(red: 0.612, green: 0.157, blue: 0.694)

    static let cardBackground = Color.white
    static let screenBackgroundFallback = Color(red: 0.976, green: 0.906, blue: 0.965)

    static let freeTag = Color(red: 0.549, green: 0.792, blue: 0.396)
    static let premiumTag = Color(red: 0.976, green: 0.702, blue: 0.235)

    // MARK: - Fonts page (docs/figma/5.png)

    /// #77A509 on #F4F6D2 — the "Free" badge on a font card. Sampled off 5.png; deliberately a
    /// separate pair from `freeTag`, which is a bare text colour with no chip behind it.
    static let freeChipText = Color(red: 119 / 255, green: 165 / 255, blue: 9 / 255)
    static let freeChipBackground = Color(red: 244 / 255, green: 246 / 255, blue: 210 / 255)
    /// #FD981B on #FDEDC6 — the "Pro" counterpart.
    static let proChipText = Color(red: 253 / 255, green: 152 / 255, blue: 27 / 255)
    static let proChipBackground = Color(red: 253 / 255, green: 237 / 255, blue: 198 / 255)

    /// #8A8585 — the Fonts header subtitle, the "Type something..." placeholder and the
    /// apply-panel subtitle. A *warm* grey; Community's `textMuted` is a flat neutral #AAAAAA and
    /// reads visibly cooler against this page's pink background.
    static let textGreyWarm = Color(red: 138 / 255, green: 133 / 255, blue: 133 / 255)

    /// #F9D0ED — the small "Aa" chip that sits left of the "Fonts" page title.
    static let badgePink = Color(red: 249 / 255, green: 208 / 255, blue: 237 / 255)

    /// The back button's circle: a pink-to-orchid ramp, unlike the search button beside it which
    /// is flat `logoSolid`.
    static let backButtonStart = Color(red: 227 / 255, green: 171 / 255, blue: 244 / 255)
    static let backButtonEnd = Color(red: 201 / 255, green: 121 / 255, blue: 224 / 255)

    // MARK: - Themes page (docs/figma/8.png)

    /// #A92CC0 — the arrow-into-tray mark on each theme tile. Sampled at the glyph core; it is a
    /// touch lighter than `logoSolid`, which every *stroke* on this page uses.
    static let downloadGlyph = Color(red: 169 / 255, green: 44 / 255, blue: 192 / 255)

    // MARK: - Create Custom Theme (docs/figma/4.png)

    /// #BD16F7 — the hue both saturation/value squares are built on. Solved from the squares
    /// rather than guessed: white blended toward this colour across the square's width, then
    /// darkened down its height, reproduces every sample taken off the KEY COLOR rect to within a
    /// couple of levels. It is markedly bluer than `purple`, which reads too red in the ramp.
    static let pickerHue = Color(red: 189 / 255, green: 22 / 255, blue: 247 / 255)

    // MARK: - Profile (docs/figma/3.png)

    /// #621570 — the "Edit Profile" capsule's 2px outline. Markedly darker than the label inside
    /// it, which is a separate colour: the first build set both to this and the text came out
    /// visibly muddy against the design's. Sampled at the stroke's core on the top and left edges.
    static let editProfileStroke = Color(red: 98 / 255, green: 21 / 255, blue: 112 / 255)

    /// #9012A7 — the "Edit Profile" label and the pencil beside it. Close to `logoSolid` (#9C28B1)
    /// but a touch deeper, and definitely not the capsule's own stroke colour.
    static let editProfileInk = Color(red: 144 / 255, green: 18 / 255, blue: 167 / 255)

    /// The RECENT grid, read left-to-right then top-to-bottom off the six swatches.
    static let recentSwatches: [Color] = [
        Color(red: 241 / 255, green: 191 / 255, blue: 243 / 255),
        Color(red: 236 / 255, green: 159 / 255, blue: 212 / 255),
        Color(red: 161 / 255, green: 201 / 255, blue: 234 / 255),
        Color(red: 221 / 255, green: 172 / 255, blue: 215 / 255),
        Color(red: 240 / 255, green: 194 / 255, blue: 211 / 255),
        Color(red: 182 / 255, green: 137 / 255, blue: 208 / 255)
    ]
}

enum MochiGradient {
    static let background = LinearGradient(
        colors: [
            Color(red: 0.988, green: 0.851, blue: 0.925),
            Color(red: 0.902, green: 0.784, blue: 0.949),
            Color(red: 0.804, green: 0.741, blue: 0.961)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    static let primaryButton = LinearGradient(
        colors: [MochiColor.pink, MochiColor.pink.opacity(0.85), MochiColor.purple],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// Home's action-card buttons AND the selected FONTS/THEMES pill — one shared ramp, sampled
    /// across the pill in docs/figma/1.png. It is not a straight pink-to-purple two-stop: it starts
    /// orchid (206,118,219), warms to pink about a third of the way across (226,127,204), then
    /// runs to periwinkle (143,124,233). A two-stop approximation loses that warm middle and reads
    /// noticeably flatter.
    static let softButton = LinearGradient(
        stops: [
            .init(color: Color(red: 206/255, green: 118/255, blue: 219/255), location: 0.0),
            .init(color: Color(red: 226/255, green: 127/255, blue: 204/255), location: 0.32),
            .init(color: Color(red: 143/255, green: 124/255, blue: 233/255), location: 1.0)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// The Fonts page's accent ramp — the selected category pill, both Apply buttons, the "Aa"
    /// title chip and the selected tab icon all share it. Sampled across the "All" pill in
    /// docs/figma/5.png: it starts a full step pinker than `softButton` (#DE79D3 against #CE76DB)
    /// and ends slightly deeper in periwinkle, which is why it is its own ramp rather than a reuse.
    static let fontsAccent = LinearGradient(
        stops: [
            .init(color: Color(red: 222/255, green: 121/255, blue: 211/255), location: 0.0),
            .init(color: Color(red: 225/255, green: 127/255, blue: 206/255), location: 0.28),
            .init(color: Color(red: 135/255, green: 127/255, blue: 233/255), location: 1.0)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    static let logoText = LinearGradient(
        colors: [MochiColor.purpleDark, MochiColor.pink],
        startPoint: .leading,
        endPoint: .trailing
    )

    // MARK: - Themes page ramps (docs/figma/8.png)

    /// The Themes page's "Apply" capsule and its selected "All" pill. Sampled across the 236px
    /// Apply button at 8px intervals: it is NOT `softButton`. That ramp opens orchid (206,118,219)
    /// and only warms to pink a third of the way in; this one *starts* pink (229,128,201), holds
    /// that through the first 30%, then runs to periwinkle. Using `softButton` here left the
    /// button's left end visibly too purple.
    static let themeButton = LinearGradient(
        stops: [
            .init(color: Color(red: 229/255, green: 128/255, blue: 201/255), location: 0.0),
            .init(color: Color(red: 227/255, green: 127/255, blue: 204/255), location: 0.30),
            .init(color: Color(red: 208/255, green: 120/255, blue: 221/255), location: 0.55),
            .init(color: Color(red: 143/255, green: 123/255, blue: 233/255), location: 1.0)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// The two header discs. Both the back *and* the search button carry this ramp with a black
    /// glyph — unlike the Fonts frame, where the search button is a flat `logoSolid` disc with a
    /// white glyph. A horizontal slice through either circle spans only part of `themeButton`'s
    /// range, so it gets its own shorter ramp rather than reusing that one.
    static let themeCircleButton = LinearGradient(
        colors: [
            Color(red: 226/255, green: 128/255, blue: 204/255),
            Color(red: 188/255, green: 121/255, blue: 227/255)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// The rounded-square chip holding the palette mark left of the "Themes" title.
    static let themeBadge = LinearGradient(
        colors: [
            Color(red: 222/255, green: 121/255, blue: 216/255),
            Color(red: 161/255, green: 126/255, blue: 231/255)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    // MARK: - Create Custom Theme ramps (docs/figma/4.png)

    /// The selected editor pill ("Fonts"). Only 294px wide, so it crosses a much shorter stretch of
    /// the family's pink→periwinkle run than `softButton` does — sampled end to end it opens on
    /// (229,129,197) and closes on (177,123,227), never reaching either of `softButton`'s extremes.
    static let editorPill = LinearGradient(
        colors: [
            Color(red: 229 / 255, green: 129 / 255, blue: 197 / 255),
            Color(red: 177 / 255, green: 123 / 255, blue: 227 / 255)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// The four tag chips. A touch cooler at the closing end than `editorPill`.
    static let tagPill = LinearGradient(
        colors: [
            Color(red: 228 / 255, green: 126 / 255, blue: 203 / 255),
            Color(red: 160 / 255, green: 124 / 255, blue: 229 / 255)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// Fill of the four small key-shape previews — pale lavender at the cap, deepening to orchid
    /// at the foot.
    static let keyShapePreview = LinearGradient(
        colors: [
            Color(red: 236 / 255, green: 226 / 255, blue: 247 / 255),
            Color(red: 206 / 255, green: 170 / 255, blue: 232 / 255)
        ],
        startPoint: .top,
        endPoint: .bottom
    )

    /// The rail under each saturation/value square. Figma stops the sweep at magenta rather than
    /// carrying it back round to red — its right end samples (251,4,201) — so the ramp runs to
    /// 11/12 of the wheel, not a full turn.
    static let hueSpectrum = LinearGradient(
        colors: (0...11).map { Color(hue: Double($0) / 12.0, saturation: 1, brightness: 1) },
        startPoint: .leading,
        endPoint: .trailing
    )

    static let premiumBanner = LinearGradient(
        colors: [
            Color(red: 0.416, green: 0.204, blue: 0.780),
            Color(red: 0.667, green: 0.278, blue: 0.816)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
}

enum MochiRadius {
    static let card: CGFloat = 20
    static let pill: CGFloat = 999
    static let sheet: CGFloat = 28
}

enum MochiSpacing {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 16
    static let lg: CGFloat = 24
    static let xl: CGFloat = 32
}

File: android/app/src/main/java/com/mochi/keyboard/designsystem/Theme.kt
package com.mochi.keyboard.designsystem

import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Ported from ios/MochiApp/DesignSystem/Theme.swift — keep both in sync until a shared token
 * source exists. Values are pixel-sampled off docs/figma (screen exports; see the iOS file for the
 * per-color sampling notes); this file carries only the resulting hex values plus a short pointer
 * to which frame each one came from.
 */
object MochiColor {
    val purple = Color(0xFF8B5CF6)
    val purpleDark = Color(0xFF7C3AED)
    val pink = Color(0xFFEC4999)
    val pinkLight = Color(0xFFF9B7DA)
    val lavender = Color(0xFFCEBFF9)

    /** Pure black — section headers, card titles/subtitles, pill labels all sample to (0,0,0) in figma/1.png; lighter-reading text is a lighter Inter weight, not a lighter color. */
    val textPrimary = Color.Black
    val textSecondary = Color(0xFF757080)

    /** Flat #AAAAAA — Community's search placeholder, "N Themes" byline and theme descriptions (figma/2.png). */
    val textMuted = Color(0xFFAAAAAA)

    /** #9750AB — the "by <creator>" byline under a Top Themes card; lighter/pinker than logoSolid. */
    val creatorLink = Color(0xFF9750AB)

    /** #F44336 — the filled like heart (Material Red 500, not the app's own pink). */
    val heart = Color(0xFFF44336)

    /** #8A4FA0 — the 1pt outline on Community's search field, filter pills and round download buttons. */
    val outline = Color(0xFF8A4FA0)

    /** Sampled directly from docs/figma/1.png — the "Mochi" wordmark is flat, not a gradient. */
    val logoSolid = Color(0xFF9C28B1)

    val cardBackground = Color.White
    val screenBackgroundFallback = Color(0xFFF9E7F6)

    val freeTag = Color(0xFF8CCA65)
    val premiumTag = Color(0xFFF9B33C)

    // Fonts page (figma/5.png)
    val freeChipText = Color(0xFF77A509)
    val freeChipBackground = Color(0xFFF4F6D2)
    val proChipText = Color(0xFFFD981B)
    val proChipBackground = Color(0xFFFDEDC6)

    /** #8A8585 — a warm grey; Community's textMuted is neutral #AAAAAA and reads cooler against Fonts' pink background. */
    val textGreyWarm = Color(0xFF8A8585)
    val badgePink = Color(0xFFF9D0ED)
    val backButtonStart = Color(0xFFE3ABF4)
    val backButtonEnd = Color(0xFFC979E0)

    // Themes page (figma/8.png)
    val downloadGlyph = Color(0xFFA92CC0)

    // Create Custom Theme (figma/4.png)
    val pickerHue = Color(0xFFBD16F7)

    // Profile (figma/3.png)
    val editProfileStroke = Color(0xFF621570)
    val editProfileInk = Color(0xFF9012A7)

    /** The RECENT grid, left-to-right then top-to-bottom off the six swatches. */
    val recentSwatches = listOf(
        Color(0xFFF1BFF3),
        Color(0xFFEC9FD4),
        Color(0xFFA1C9EA),
        Color(0xFFDDACD7),
        Color(0xFFF0C2D3),
        Color(0xFFB689D0)
    )
}

object MochiGradient {
    val background = Brush.linearGradient(
        colors = listOf(Color(0xFFFCD9EC), Color(0xFFE6C8F2), Color(0xFFCDBDF5))
    )

    val primaryButton = Brush.horizontalGradient(
        colors = listOf(MochiColor.pink, MochiColor.pink.copy(alpha = 0.85f), MochiColor.purple)
    )

    /** Home's action-card buttons and the selected FONTS/THEMES pill — starts orchid, warms to pink a third of the way across, ends periwinkle (figma/1.png). */
    val softButton = Brush.horizontalGradient(
        colorStops = arrayOf(
            0.0f to Color(0xFFCE76DB),
            0.32f to Color(0xFFE27FCC),
            1.0f to Color(0xFF8F7CE9)
        )
    )

    /** Fonts page accent ramp — the selected category pill, both Apply buttons, the "Aa" title chip and the selected tab icon (figma/5.png). A full step pinker at the start than softButton. */
    val fontsAccent = Brush.horizontalGradient(
        colorStops = arrayOf(
            0.0f to Color(0xFFDE79D3),
            0.28f to Color(0xFFE17FCE),
            1.0f to Color(0xFF877FE9)
        )
    )

    val logoText = Brush.horizontalGradient(
        colors = listOf(MochiColor.purpleDark, MochiColor.pink)
    )

    /** Themes page's Apply capsule and selected "All" pill — opens pink rather than softButton's orchid (figma/8.png). */
    val themeButton = Brush.horizontalGradient(
        colorStops = arrayOf(
            0.0f to Color(0xFFE580C9),
            0.30f to Color(0xFFE37FCC),
            0.55f to Color(0xFFD078DD),
            1.0f to Color(0xFF8F7BE9)
        )
    )

    /** The two Themes-page header discs (back + search) — both carry this ramp with a black glyph. */
    val themeCircleButton = Brush.horizontalGradient(
        colors = listOf(Color(0xFFE280CC), Color(0xFFBC79E3))
    )

    /** The rounded-square chip holding the palette mark left of the "Themes" title. */
    val themeBadge = Brush.linearGradient(
        colors = listOf(Color(0xFFDE79D8), Color(0xFFA17EE7))
    )

    /** Create Custom Theme's selected editor pill ("Fonts") — only 294px wide in figma/4.png, so it spans a shorter stretch of the family ramp than softButton. */
    val editorPill = Brush.horizontalGradient(
        colors = listOf(Color(0xFFE581C5), Color(0xFFB17BE3))
    )

    /** The four tag chips — a touch cooler at the closing end than editorPill. */
    val tagPill = Brush.horizontalGradient(
        colors = listOf(Color(0xFFE47ECB), Color(0xFFA07CE5))
    )

    /** Fill of the four key-shape previews — pale lavender at the cap, deepening to orchid at the foot. */
    val keyShapePreview = Brush.verticalGradient(
        colors = listOf(Color(0xFFECE2F7), Color(0xFFCEAAE8))
    )

    /** The rail under each saturation/value square. Figma stops the sweep at magenta rather than carrying it back to red, so this runs 11/12 of the hue wheel, not a full turn. */
    val hueSpectrum = Brush.horizontalGradient(
        colors = (0..11).map { i -> Color.hsv(hue = i * 30f, saturation = 1f, value = 1f) }
    )

    val premiumBanner = Brush.linearGradient(
        colors = listOf(Color(0xFF6A34C7), Color(0xFFAA47D0))
    )
}

object MochiRadius {
    val card = 20.dp
    val pill = 999.dp
    val sheet = 28.dp
}

object MochiSpacing {
    val xs = 4.dp
    val sm = 8.dp
    val md = 16.dp
    val lg = 24.dp
    val xl = 32.dp
}
